topics/user_interface_components/status_bar_widgets.md

```diff
@@ -0,0 +1,114 @@
+To control the disposing of a widget, implement the `disposeWidget()`, if you just want to dispose it, use `Disposer.dispose(widget)`.
+
+Any widget should implement the
+[`StatusBarWidget`](upsource:///platform/platform-api/src/com/intellij/openapi/wm/StatusBarWidget.java)
+interface.
+
+To reuse the IntelliJ Platform implementation, you can extend one of two classes:
+
+- [`EditorBasedWidget`](upsource:///platform/platform-api/src/com/intellij/openapi/wm/impl/status/EditorBasedWidget.java)
+- [`EditorBasedStatusBarPopup`](upsource:///platform/platform-api/src/com/intellij/openapi/wm/impl/status/EditorBasedStatusBarPopup.java)
+
+## EditorBasedWidget
+
+`EditorBasedWidget` is the basic widget implementation.
+To implement it, override `ID()` where returns the unique ID of the widget.
+This identifier may be needed to later get a widget instance.
+
+The IntelliJ Platform provides several predefined widget appearance options:
+
+- `com.intellij.openapi.wm.StatusBarWidget.IconPresentation`
+
+  Widget with only an icon.
+
+  Example:
+  [PowerSaveStatusWidgetFactory](upsource:///platform/platform-impl/src/com/intellij/openapi/wm/impl/status/PowerSaveStatusWidgetFactory.java)
+
+- `com.intellij.openapi.wm.StatusBarWidget.TextPresentation`
+
+  Widget with only a text.
+
+  Example:
+  [PositionPanel](upsource:///platform/platform-impl/src/com/intellij/openapi/wm/impl/status/PositionPanel.java)
+
+- `com.intellij.openapi.wm.StatusBarWidget.MultipleTextValuesPresentation`
+
+  Widget with a text and a popup.
+
+  Example:
+  [DvcsStatusWidget](upsource:///platform/dvcs-impl/src/com/intellij/dvcs/ui/DvcsStatusWidget.java)
+
+> Note that they can't be combined to get, for example, an icon and a text.
+>
+{type="note"}
+
+To use the selected appearance, return a class that implements one of the above interfaces from `getPresentation()`.
+
+To create a widget with custom content, it should implement the
+[`CustomStatusBarWidget`](upsource:///platform/platform-api/src/com/intellij/openapi/wm/CustomStatusBarWidget.java)
+interface.
+Override `getComponent()` to return the custom widget's component to display.
+
+Example:
+[MemoryUsagePanel](upsource:///platform/platform-impl/src/com/intellij/openapi/wm/impl/status/MemoryUsagePanel.java)
+
+## EditorBasedStatusBarPopup
+
+`EditorBasedStatusBarPopup` is the basis for all widgets that have a popup with a list of actions.
+For example, the encoding widget of the current file.
+
+First, implement `createComponent()`, where returns a component for display.
+Each update of the widget IDE calls `updateComponent()` to update this component.
+In `updateComponent()` implementation, you can describe how the widget should change depending on the current state.
+
+Implement `getWidgetState()` where returns the current state of the widget.
+This state will be passed to the `updateComponent()` when the widget is updated.
+The method accepts a file that's now open in the editor.
```

Comment:
that's now open -> that is currently opened in the editor

Reply:
Fixed.